### PR TITLE
fix: use datasets for transcripts and handle live stream links

### PR DESCRIPTION
**Pull request by njhale**
- Use datasets for transcripts instead of workspace files
- Handle live stream links
- Improve output for bad links and disabled transcript errors
- Clean up transcript text with an "internal" tool call
- Drop fallback to the advanced transcriber until it's more polished

Addresses: https://github.com/otto8-ai/otto8/issues/435 and https://github.com/otto8-ai/otto8/issues/436

### CODE REVIEW

**Review by iwilltry42 — APPROVED** [review]
If it's crucial I guess this could be replicated in code in https://github.com/otto8-ai/tools/blob/main/google/youtube/helpers.py as well as a second "safety net" if the LLM still decides to call the tool, right?

File: google/youtube/tool.gpt [hunk 1]
```diff
@@ -6,35 +6,24 @@ Share Tools: Transcribe YouTube Video
...
-Write the corrected text to a file in the $GPTSCRIPT_WORKSPACE_DIR with a filename based on the VIDEO_ID.
-Let the user know where the file is located, and give a short 2 sentence summary of the video.
-
----
-Name: Fetch Raw YouTube Transcript 
-Metadata: index: false
 Description: Get YouTube's raw text transcript of a video.
-Param: video_url: The URL of a YouTube video. (e.g. https://www.youtube.com/watch?v=VIDEO_ID or https://youtu.be/VIDEO_ID)
+Share Context: YouTube Context
+Param: video_url: The URL of a YouTube video. (e.g. URLs in the form https://www.youtube.com/watch?v=<VIDEO_ID> or https://youtu.be/<VIDEO_ID>). Does not accept live stream URLs (e.g. URLs starting vith https://www.youtube.com/live/).
```
Comment (drpebcak) [inline google/youtube/tool.gpt:11]: I'm not sure this is the correct fix - youtube urls seem to really suck. For example, if you look at the clio intro video (https://www.youtube.com/watch?v=A3YHG-5R3YE), the share button generates a live stream url - (https://www.youtube.com/live/A3YHG-5R3YE?si=8oI2o1zBrQROjvkd). but this stream is completed, and it does now have transcripts.

File: google/youtube/tool.gpt [hunk 2]
```diff
@@ -6,35 +6,24 @@ Share Tools: Transcribe YouTube Video
...
 
 #!/usr/bin/env python3 ${GPTSCRIPT_TOOL_DIR}/transcriber.py
 
 ---
-Name: Generate Raw YouTube Transcript
```
Comment (drpebcak) [inline google/youtube/tool.gpt:30]: If we are deleting this tool outright, we should also delete the code.

File: google/youtube/tool.gpt (hunk 1 above)
Comment (njhale) [reply google/youtube/tool.gpt:11]: Yeah, I'm looking into some alternatives.

Comment (drpebcak) [reply google/youtube/tool.gpt:11]: I think you would be better off just bubbling up the error that transcripts don't exist for the video - that will happen sometimes with regular youtube videos as well.

**Comment by njhale** [comment]
This PR has changed pretty significantly. Putting back into draft while I test the changes.

**Comment by njhale** [comment]
Working and ready for review.

File: google/youtube/tool.gpt (hunk 1 above)
Comment (njhale) [reply google/youtube/tool.gpt]: Done!

File: google/youtube/tool.gpt (hunk 2 above)
Comment (njhale) [reply google/youtube/tool.gpt:30]: Done!

File: google/youtube/helpers.py [hunk 3]
```diff
@@ -1,18 +1,102 @@
...
+    text: str
+
+
+gptscript_client = GPTScript()
+
+
+async def get_clean_transcript(text: str) -> Transcript:
+    model = os.getenv('MODEL')
+    if model is None or model == '':
+        model = 'gpt-4o'
```
Comment (drpebcak) [inline google/youtube/helpers.py:53]:
```suggestion
    model = os.getenv('MODEL', 'gpt-4o')
```

**Review by drpebcak — CHANGES_REQUESTED** [review]

**Review by g-linville — APPROVED** [review]

File: google/youtube/helpers.py [hunk 4]
```diff
@@ -1,18 +1,102 @@
...
+        jsonResponse=True,
+        instructions=dedent(f'''
+            Clean up the TEXT below by adding punctuation and correcting all misspellings and errors.
+            Do not truncate or reword any of the TEXT, just add punctuation and correct mistakes.
+            Afterwards, write a two sentence summary of the transcript and return an object containing the clean text and summary with the following JSON format:
+
+            {{
+                "summary": "<two sentence summary>",
+                "text": "<clean text>"
+            }}
```
[the remark this answers is not in the document]
Comment (StrongMonkey) [reply google/youtube/helpers.py:69]: because it is python? :)

**Review by StrongMonkey — APPROVED** [review]

File: google/youtube/helpers.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (njhale) [reply google/youtube/helpers.py:69]: This is in an f-string, and `{` triggers interpolation; the extra `{` escapes it

File: google/youtube/helpers.py (hunk 3 above)
Comment (njhale) [reply google/youtube/helpers.py]: Done

**Pull request merged**